src/modules/fancyzones/editor/FancyZonesEditor/CanvasZone.xaml.cs

```diff
@@ -16,192 +17,350 @@ namespace FancyZonesEditor
+                        key_positions.Add(curr_p);
+                        key_positions.Add(curr_p + curr_w);
+                        if (mode == ModeBoth)
+                        {
+                            key_positions.Add(curr_p - track_w);
+                            key_positions.Add(curr_p + curr_w - track_w);
+                        }
+                    }
+                }
 
-            rect.X += (int)xDelta;
-            rect.Y += (int)yDelta;
+                // Remove duplicates and sort
+                key_positions.Sort();
+                Snaps = new List<int>();
+                if (key_positions.Count > 0)
+                {
+                    Snaps.Add(key_positions[0]);
+                    for (int i = 1; i < key_positions.Count; ++i)
+                    {
+                        if (key_positions[i] != key_positions[i - 1])
+                        {
+                            Snaps.Add(key_positions[i]);
+                        }
+                    }
+                }
 
-            Canvas.SetLeft(this, rect.X);
-            Canvas.SetTop(this, rect.Y);
-            Model.Zones[ZoneIndex] = rect;
+                // Initialize minValue
+                if (mode == ModeLow)
+                {
+                    // We're dragging the low edge, don't go below zero
+                    MinValue = 0;
+                }
+                else if (mode == ModeHigh)
+                {
+                    // We're dragging the high edge, don't make the zone smaller than min_w
+                    MinValue = track_p + min_w;
+                }
+                else if (mode == ModeBoth)
+                {
+                    // We're moving the window, don't move it below zero
+                    MinValue = 0;
+                }
+
+                // Initialize maxValue
+                if (mode == ModeLow)
+                {
+                    // We're dragging the low edge, it can't make the zone smaller than min_w
+                    MaxValue = track_p + track_w - min_w;
+                }
+                else if (mode == ModeHigh)
+                {
+                    // We're dragging the high edge, don't go off the screen
+                    MaxValue = screen_w;
+                }
+                else if (mode == ModeBoth)
+                {
+                    // We're moving the window, don't go off the screen (this time the lower edge is tracked)
+                    MaxValue = screen_w - track_w;
+                }
+
+                // Initialize position
+                if (mode == ModeLow)
+                {
+                    Position = track_p;
+                }
+                else if (mode == ModeHigh)
+                {
+                    Position = track_p + track_w;
+                }
+                else if (mode == ModeBoth)
+                {
+                    Position = track_p;
+                }
```

Comment:
Wouldn't it be better if this was i single switch/case operation?

Reply:
Agreed, I added a commit.